### PR TITLE
fix: download apptainer images without docker client

### PR DESCRIPTION
**Pull request by mhhennig**
Previous code failed when docker could not be run on the host system. Now getting the docker image via spython.

### CODE REVIEW

**Comment by mhhennig** [comment]
Btw, this should solve #3221

File: src/spikeinterface/sorters/external/kilosort4.py [hunk 1]
```diff
@@ -232,7 +233,7 @@ def _run_from_folder(cls, sorter_output_folder, params, verbose):
                 do_CAR=do_CAR,
                 invert_sign=invert_sign,
                 device=device,
-                save_preprocesed_copy=save_preprocessed_copy,  # this kwarg is correct (typo)
+                save_preprocessed_copy=save_preprocessed_copy,  # this kwarg is correct (typo)
```
Comment (alejoe91) [inline src/spikeinterface/sorters/external/kilosort4.py:236]: @mhhennig this was a typo in a previous KS version (see https://github.com/SpikeInterface/spikeinterface/pull/3316). Can you revert the KS changes? It's better to have a dedicated PR to handle them

**Review by alejoe91 — CHANGES_REQUESTED** [review]
See comment

**Comment by mhhennig** [comment]
Sorry, this ended up in the same PR. So here we now have:
- fix for failing to pull containers when only singularity is available
- fix for recent changed in Kilosort4 API (latest pypi version (4.0.16 from 14 August) which prevents it from running; note this change will break older versions, but the code is not compatible

File: src/spikeinterface/sorters/external/kilosort4.py (hunk 1 above)
Comment (mhhennig) [reply src/spikeinterface/sorters/external/kilosort4.py:236]: Ok

**Comment by mhhennig** [comment]
Removed the KS commit.

**Review by alejoe91 — APPROVED** [review]
Thansk @mhhennig 

It's working on my end too after uninstallinf docker!

File: src/spikeinterface/sorters/container_tools.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ def __init__(self, mode, container_image, volumes, py_user_base, extra_kwargs):
                 singularity_image = sif_file
             else:
 
-                docker_image = self._get_docker_image(container_image)
+                docker_image = Client.load("docker://" + container_image)
```
Comment (samuelgarcia) [inline src/spikeinterface/sorters/container_tools.py:102]: The patch should have been done in `_get_docker_image`

**Pull request merged**